libraries/ms-common/src/x86/flash.c

```diff
@@ -0,0 +1,87 @@
+#include "flash.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include "log.h"
+
+#define FLASH_FILENAME "x86_flash"
+
+static FILE *s_flash_fp = NULL;
+
+StatusCode flash_init(void) {
+  if (s_flash_fp != NULL) {
+    fclose(s_flash_fp);
+  }
+
+  s_flash_fp = fopen(FLASH_FILENAME, "r+b");
+  if (s_flash_fp == NULL) {
+    LOG_DEBUG("Setting up new flash file\n");
+    s_flash_fp = fopen(FLASH_FILENAME, "w+b");
+    for (int i = 0; i < NUM_FLASH_PAGES; i++) {
+      flash_erase(i);
+    }
+  }
+
+  return STATUS_CODE_OK;
+}
+
+StatusCode flash_read(uintptr_t address, size_t read_bytes, uint8_t *buffer, size_t buffer_len) {
+  if (buffer_len < read_bytes || address < FLASH_BASE_ADDR ||
+      (address + read_bytes) > FLASH_END_ADDR) {
+    return status_code(STATUS_CODE_OUT_OF_RANGE);
+  }
+
+  fseek(s_flash_fp, (intptr_t)address, SEEK_SET);
```

Comment:
If `uintptr_t` -> `intptr_t` it is possible to fseek backwards... I don't think the numbers we have now would allow this but just something we might want to note in `x86/flash_mcu.h`

Reply:
The only reason I do this is because otherwise the compiler complains. I can cast it first and make sure it's above 0?